src/hdmf/common/resources.py

```diff
@@ -405,6 +407,78 @@ def _get_file_from_container(self, **kwargs):
                 msg = 'Could not find file. Add container to the file.'
                 raise ValueError(msg)
 
+    @docval({'name': 'file',  'type': ExternalResourcesManager, 'doc': 'The file associated with the container.',
+             'default': None},
+            {'name': 'container', 'type': (str, AbstractContainer), 'default': None,
+             'doc': ('The Container/Data object that uses the key or '
+                     'the object_id for the Container/Data object that uses the key.')},
+            {'name': 'attribute', 'type': str,
+             'doc': 'The attribute of the container for the external reference.', 'default': None},
+            {'name': 'field', 'type': str, 'default': '',
+             'doc': ('The field of the compound data type using an external resource.')},
+            {'name': 'key', 'type': (str, Key), 'default': None,
+             'doc': 'The name of the key or the Key object from the KeyTable for the key to add a resource for.'},
+            {'name': 'term_set', 'type': TermSet, 'default': None,
+             'doc': 'The TermSet to be used if the container/attribute does not have one.'}
+            )
+    def add_ref_term_set(self, **kwargs):
+        file = kwargs['file']
+        container = kwargs['container']
+        attribute = kwargs['attribute']
+        key = kwargs['key']
+        field = kwargs['field']
+        term_set = kwargs['term_set']
```

Comment:
Can you please add tests for TermSet. It looks like there are many uncovered lines.

Reply:
Some of the code cov lines don't make sense. Some ask for tests to cover the setting of the kwargs, while some are covered by tests but it doesn't pick up.

Reply:
I believe this is due to the skip.

Reply:
Or are the tests not being run on GitHub because LinkML is not installed?

Reply:
Most likely. We left it as an optional dependency per Ryan's advice of linkml being a heavy dependency.